native/src/components/__tests__/EmbeddedOffers.spec.tsx

```diff
@@ -7,21 +7,21 @@ import { CategoryModel, OfferModel } from 'shared/api'
 import render from '../../testing/render'
 import EmbeddedOffers from '../EmbeddedOffers'
 
-jest.mock('../SprungbrettOffer.tsx', () => ({
-  __esModule: true,
-  default: ({ sprungbrettOffer }: { sprungbrettOffer: OfferModel }) => {
+jest.mock('../SprungbrettOffer.tsx', () => {
+  const MockedSprungbrettOffer = ({ sprungbrettOffer }: { sprungbrettOffer: OfferModel }) => {
     const { Text } = require('react-native')
     return <Text>This is the Sprungbrett offer called {sprungbrettOffer.title}</Text>
-  },
-}))
+  }
+  return MockedSprungbrettOffer
```

Comment:
I honestly find it more legible than a function to a function, but I don't feel strongly about it. I can also change it if you want me to 🤷

Reply:
I don't really have a preference on the specific code style either. I just think it should generally be uniform and consistent. I don't know though why the linter does not complain though...

Reply:
And my IDE highlights this (which I don't like... and yes, I could just turn it off but I think generally it is useful) xD
![image](https://github.com/user-attachments/assets/9003d14c-eae4-403a-b576-2b43813a0498)

Reply:
I think the linter skips test files. Or at least it does in my environment 🤔  I assumed it's on purpose?

Reply:
Some rules are definitely skipped for test files, but not this one. The no redundant local variable is just an IDE thing (see above) though. And since there is another thing than just a return in this function body, the linting rule does not complain.

Reply:
Ooooh, of course. Hmm. I'm still in favour of keeping it with the extra variable just because I find functions returning functions somewhat confusing but at the same time, it's just a test file, nobody really needs to be able to understand how exactly the mocking worked here. I also definitely feel your pain of the linter marking the file as problematic, even if it's only the local one. Well, I think I just convinced myself 😂